### DISCUSSION TITLE
Replace .gif flag icons to .svg icons

### DISCUSSION THREAD

**Discussion by Sulpher** [joomla/joomla-cms#47128] [2. Cold Features]
Well, the flag icons used in the language switcher module still are in old gif format which is not scalable. 
I suggest replacing old icons with the new SVG icons taken from the free package.
Such an approach will give the ability to customize its size or change anything via CSS.

Example of existing flag icon:
<img width="830" alt="Снимок экрана 2021-09-13 в 17 43 30" src="https://user-images.githubusercontent.com/3432048/133104816-b7851929-78fb-4318-9c3c-ccfed2ae90da.png">

The flags can be bigger and scalable:
<img width="1180" alt="Снимок экрана 2021-09-13 в 17 42 44" src="https://user-images.githubusercontent.com/3432048/133104627-b3f8353d-0436-4cf4-9d13-47a46d8d0923.png">

Here is the package:
https://flagicons.lipis.dev/
https://github.com/lipis/flag-icon-css/blob/master/LICENSE
MIT license

**Comment by dgrammatiko** [comment]
FYI https://github.com/joomla/joomla-cms/pull/21410

**Comment by simbus82** [comment]
Maybe after 3 years, those who have slowed down the evolution of Joomla, now they have changed their minds and we will finally have some decent assets!
I get sick when I see gifs in a website UI.
Thanks for this issue @Sulpher

In our websites we typically do an [override ](https://github.com/joomla/joomla-cms/pull/21410#issuecomment-424010160)choosing carefully the icons in SVG. It would be nice if they were in the core though...

**Comment by Sulpher** [comment]
> In our websites we typically do an [override ](https://github.com/joomla/joomla-cms/pull/21410#issuecomment-424010160)choosing carefully the icons in SVG. It would be nice if they were in the core though...

Thanks for sharing the override layout, Simone.

**Comment by sozzled** [comment]
Interesting suggestion.  I read the [very old] discussion [that went nowhere].  Several commentators noted the distinction between country flags and languages spoken in those countries.  For instance, Switzerland has one flag but four official languages.

While country flag images may **_suggest_** to people visiting a website that the content is available in more than one language, the use of a national flag does not imply that transnational languages are implemented on the website.  For instance, I would never use the Union Flag (the official flag of the United Kingdom) on any of my websites.  As a proud Australian, I would display the Australian flag (perhaps) ... but so what?  Flags do not mean languages.  Worse, flag images are useless to people who have visual impairment (even if you used an "alt=" tag.

As I say, it's an interesting suggestion.  I think it's a good idea if J! offers the ability to use SVG instead of GIF.  Some people will probably be attracted to use scalable graphics instead of "ugly" GIFs.  That's about as far as I am prepared to go, though.  Offer the choice, yes; implement the feature at one's own discretion (or risk, or peril).  :)

**Comment by brianteeman** [comment]
Users already have the choice

**Comment by sozzled** [comment]
No dispute there, Brian. :sunglasses:

**Comment by Sulpher** [comment]
> Interesting suggestion. I read the [very old] discussion [that went nowhere]. Several commentators noted the distinction between country flags and languages spoken in those countries. For instance, Switzerland has one flag but four official languages.
> 
> While country flag images may **_suggest_** to people visiting a website that the content is available in more than one language, the use of a national flag does not imply that transnational languages are implemented on the website. For instance, I would never use the Union Flag (the official flag of the United Kingdom) on any of my websites. As a proud Australian, I would display the Australian flag (perhaps) ... but so what? Flags do not mean languages. Worse, flag images are useless to people who have visual impairment (even if you used an "alt=" tag.
> 
> As I say, it's an interesting suggestion. I think it's a good idea if J! offers the ability to use SVG instead of GIF. Some people will probably be attracted to use scalable graphics instead of "ugly" GIFs. That's about as far as I am prepared to go, though. Offer the choice, yes; implement the feature at one's own discretion (or risk, or peril). :)

I clearly understand what you mean but guess this is the question of a webmaster's choice according to the situation: the country, language and target auditory.
In countries like Switzerland, users can use abbreviations like DE, IT, FR and  Romansch (I do not know its 2-letter abbreviation) or full language names, the same goes with other countries. So, let the site owner decide the way to go but technically we already have such an ability, and let just replace gif icons to svg icons. 
Especially, @simbus82 already suggested code changes and the icon pack mentioned in this discussion above has MIT license, so it should not be hard, just need to be approved by responsible persons.

**Comment by sozzled** [comment]
A little off-topic:  I've used "emoji flags" to display country information.  Because emojis already exist (and they're reasonably scalable), there may be no need to create a package of images (in GIF, SVG or other format).  Just saying.

**Comment by simbus82** [comment]
> A little off-topic: I've used "emoji flags" to display country information. Because emojis already exist (and they're reasonably scalable), there may be no need to create a package of images (in GIF, SVG or other format). Just saying.

Emoji are not assets, they are an interpretation made by the OS, and they aren't accessible.
IMHO here we have only to change old .gif with svg versions, no matter about country/language debate (it is a subject to be evaluated at a later time)

**Comment by Giuse69** [comment]
Hi, I add that current GIFs are considered "bad" SEO practice by Google, since on mobile they have a too low resolution. I override the mod_language to load at least custom png images with double size (36x24) while setting width still to 18 and height to 12 so that flags keep the same size as now but as a higher definition icons. The improvement in quality is also visible on desktop.
So I'm interested to see this issue proceed with a more modern visualization of flags.
thanks

**Comment by joomleb** [comment]
> ...GIFs are considered "bad" SEO practice by Google... I'm interested to see this issue proceed with a more modern visualization of flags....

+1

**Comment by mckillo** [comment]
...GIFs are considered "bad" SEO practice by Google... I'm interested to see this issue proceed with a more modern visualization of flags....

+1 

I agree too. J4 needs an update of flags icon, and "it's a good idea if J! offers the ability to use SVG instead of GIF"
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35554">issues.joomla.org/tracker/joomla-cms/35554</a>.</sub>

**Comment by joomlafun** [comment]
Even though we are lucky to have a core multilanguage feature which is not the case with wordpress. The default quality of those Gifs make your site look amateuristic and you recognize a Joomla website straight ahead when you see a pixelated english language icon.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35554">issues.joomla.org/tracker/joomla-cms/35554</a>.</sub>

**Comment by joomlafun** [comment]
Hi, 4 years later the english default flag on joomla still looks like this, please fix this.

<img width="189" height="69" alt="Image" src="https://github.com/user-attachments/assets/52cf0e06-82c8-4acb-bde8-61212cf63495" />
